File: articles/grit-and-long-term-achievement/go/grit_long_term_achievement_score.go
package main

import "fmt"

func GritScore(perseveranceEffort, consistencyInterests float64) float64 {
	return 0.60*perseveranceEffort + 0.40*consistencyInterests
}

func AchievementReadinessIndex(
	grit,
	deliberatePractice,
	feedbackQuality,
	socialSupport,
	opportunityAccess,
	burnout float64,
) float64 {
	return 0.18*grit +
		0.28*deliberatePractice +
		0.18*feedbackQuality +
		0.18*socialSupport +
		0.22*opportunityAccess -
		0.16*burnout
}

func main() {
	grit := GritScore(0.82, 0.64)
	readiness := AchievementReadinessIndex(grit, 0.86, 0.61, 0.58, 0.70, -0.28)

	fmt.Printf("Synthetic grit score: %.3f\n", grit)
	fmt.Printf("Synthetic achievement readiness index: %.3f\n", readiness)
}

File: articles/grit-and-long-term-achievement/rust/grit_long_term_achievement_score.rs
fn grit_score(perseverance_effort: f64, consistency_interests: f64) -> f64 {
    0.60 * perseverance_effort + 0.40 * consistency_interests
}

fn achievement_readiness_index(
    grit: f64,
    deliberate_practice: f64,
    feedback_quality: f64,
    social_support: f64,
    opportunity_access: f64,
    burnout: f64,
) -> f64 {
    0.18 * grit
        + 0.28 * deliberate_practice
        + 0.18 * feedback_quality
        + 0.18 * social_support
        + 0.22 * opportunity_access
        - 0.16 * burnout
}

fn main() {
    let grit = grit_score(0.82, 0.64);
    let readiness = achievement_readiness_index(grit, 0.86, 0.61, 0.58, 0.70, -0.28);

    println!("Synthetic grit score: {:.3}", grit);
    println!("Synthetic achievement readiness index: {:.3}", readiness);
}
